core/src/loader.rs

```diff
@@ -350,6 +350,87 @@ impl<'gc> LoadManager<'gc> {
         loader.movie_loader(player, request, loader_url)
     }
 
+    async fn wait_for_full_response(
+        response: OwnedFuture<Box<dyn SuccessResponse>, ErrorResponse>,
+    ) -> Result<(Vec<u8>, String, u16, bool), ErrorResponse> {
+        let response = response.await?;
+        let url = response.url().to_string();
+        let status = response.status();
+        let redirected = response.redirected();
+        let body = response.body().await;
+
+        match body {
+            Ok(body) => Ok((body, url, status, redirected)),
+            Err(error) => Err(ErrorResponse { url, error }),
+        }
+    }
+
+    pub fn load_asset_movie(
+        player: Weak<Mutex<Player>>,
+        request: Request,
+        importer_movie: Arc<SwfMovie>,
+    ) -> OwnedFuture<(), Error> {
+        let player = player
+            .upgrade()
+            .expect("Could not upgrade weak reference to player");
+
+        Box::pin(async move {
+            let fetch = player.lock().unwrap().navigator().fetch(request);
+
+            match Self::wait_for_full_response(fetch).await {
+                Ok((body, url, _status, _redirected)) => {
+                    let content_type = ContentType::sniff(&body);
+                    tracing::info!("Loading imported movie: {:?}", url);
+                    match content_type {
+                        ContentType::Swf => {
+                            let movie = SwfMovie::from_data(&body, url.clone(), Some(url.clone()))
+                                .expect("Could not load movie");
+
+                            let _is_movie_as3 = movie.is_action_script_3();
+
+                            let movie = Arc::new(movie);
+
+                            player.lock().unwrap().mutate_with_update_context(|uc| {
+                                let stage_domain = uc.avm2.stage_domain();
+                                let mut activation =
+                                    Avm2Activation::from_domain(uc.reborrow(), stage_domain);
```

Comment:
If the loaded movie isn't AS3, shouldn't we be working with an AVM1 activation? I don't think you've written an AVM1 version of the import assets code yet, and I have no clue what FP would do if an AS3 movie tried to import AS2 assets (it'd probably fail), but it should at least detect the different version and print an error or something other than blindly trying to execute with AVM2.

Reply:
As far as I know both games are AS2 not AS3. I did not find any AS3 game using Import.
But if I do the activation of the avm1 it gets stuck completely so I skipped it.